src/netaddress.cpp

```diff
@@ -723,14 +724,14 @@ bool CService::GetSockAddr(struct sockaddr* paddr, socklen_t *addrlen) const
 /**
  * @returns An identifier unique to this service's address and port number.
  */
-std::vector<unsigned char> CService::GetKey() const
-{
-     std::vector<unsigned char> vKey;
-     vKey.resize(18);
-     memcpy(vKey.data(), ip, 16);
-     vKey[16] = port / 0x100; // most significant byte of our port
-     vKey[17] = port & 0x0FF; // least significant byte of our port
-     return vKey;
+CServiceKey CService::GetKey() const
+{
+    CServiceKey vKey;
+    vKey.resize(18);
+    memcpy(vKey.data(), ip, 16);
+    vKey[16] = port / 0x100; // most significant byte of our port
+    vKey[17] = port & 0x0FF; // least significant byte of our port
+    return vKey;
```

Comment:
what about returning a `std::array<unsigned char, 18>` instead?

Reply:
I think this makes more sense

Reply:
I decided against this for now since it would require me to implement a bunch of new Serialization machinery for std::array which I couldn't figure out how to do.